SULLesson10_12/SULLesson10_12/ShopScreen/ShopScreen.swift

```diff
@@ -0,0 +1,188 @@
+        static let searchBarTrailingImageName = "line.horizontal.3.decrease"
+        static let searchText = "Search..."
+        static let currentCoutOfFurnitureText = "0"
+        static let dollarText = "$"
+    }
+
+    // MARK: - Public properties
+
+    var body: some View {
+        NavigationView {
+            contentView
+            .padding(.top)
+            .frame(maxWidth: .infinity, maxHeight: .infinity)
+            .background(LinearGradient(colors: [Color.yellow, Color.red], startPoint: .leading, endPoint: .trailing))
+        }
+    }
+
+    // MARK: - Private properties
+    
+    private let columns = [GridItem()]
+
+    @StateObject private var viewModel = ShopScreenViewModel()
+
+    private var contentView: some View {
+        ZStack {
+            VStack {
+                searchView
+                    .padding(EdgeInsets(top: 0, leading: 0, bottom: 47, trailing: 0))
+                ScrollView {
+                    furnitureAssortmentGridView
+                }
+            }
+            Text("\(Int(viewModel.totalCost))\(Constants.dollarText)")
+                .modifier(TotalCostTextModifier())
+        }
+    }
+
+    private var searchBarLeadingImageView: some View {
+        Image(systemName: Constants.searchBarLeadingImageName)
+            .resizable()
+            .modifier(SearchBarImageModifier())
+    }
+
+    private var searchBarTrailingView: some View {
+        Image(systemName: Constants.searchBarTrailingImageName)
+            .resizable()
+            .modifier(SearchBarImageModifier())
+    }
+
+    private var searchBurNavLinkView: some View {
+        NavigationLink {
+            FiltersScreen()
+        } label: {
+            searchBarTrailingView
+        }
+        .padding(.trailing)
+    }
+
+    private var searchView: some View {
+        HStack {
+            searchBarLeadingImageView
+            Text(Constants.searchText)
+                .bold()
+                .foregroundColor(.purple)
+            Spacer()
+            searchBurNavLinkView
+        }
+        .modifier(SearchBarModifier())
+    }
+
+    private var furnitureAssortmentGridView: some View {
+        LazyVGrid(columns: columns, spacing: 0) {
+            ForEach((0..<FurnitureAssortment.furnitures.count)) {
+                createFurnitureItemView(furniture: FurnitureAssortment.furnitures[$0])
+            }
+        }
+        .background(Color.white)
+    }
+
+    private var currentCountOfFurnitureView: some View {
+        ZStack {
+            RoundedRectangle(cornerRadius: 3)
+                .frame(width: 35, height: 35)
+                .foregroundColor(.white)
+                .shadow(radius: 3)
+                .padding()
+            Text(Constants.currentCoutOfFurnitureText)
+                .font(.title2)
+                .scaledToFit()
+                .bold()
+        }
+    }
+
+    // MARK: - Private methods
+
+    private func createFurnitureImageView(_ imageName: String) -> some View {
```

Comment:
```suggestion
    private func makeFurnitureImageView(_ imageName: String) -> some View {
```

все функции должны начинаться на make так по сайду написано